src/Orleans/LogConsistency/IProtocolServices.cs

```diff
@@ -0,0 +1,141 @@
+﻿
+using System;
+using System.Net;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Orleans.Providers;
+using Orleans.Runtime;
+using Orleans.MultiCluster;
+using Orleans.GrainDirectory;
+
+namespace Orleans.LogConsistency
+{
+    /// <summary>
+    /// Functionality for use by log view adaptors that use custom consistency or replication protocols.
+    /// Abstracts communication between replicas of the log-consistent grain in different clusters.
+    /// </summary>
+    public interface IProtocolServices
+    {
+        /// <summary>
+        /// Send a message to a remote cluster.
+        /// </summary>
+        /// <param name="payload">the message</param>
+        /// <param name="clusterId">the destination cluster id</param>
+        /// <returns></returns>
+        Task<IProtocolMessage> SendMessage(IProtocolMessage payload, string clusterId);
+
+
+        /// <summary>
+        /// The untyped reference for this grain.
+        /// </summary>
+        GrainReference GrainReference { get;  }
+
+        
+        /// <summary>
+        /// The multicluster registration strategy for this grain.
+        /// </summary>
+        IMultiClusterRegistrationStrategy RegistrationStrategy { get; }
+
+
+        /// <summary>
+        /// Whether this cluster is running in a multi-cluster network.
+        /// </summary>
+        /// <returns></returns>
+        bool MultiClusterEnabled { get; }
+
+
+        /// <summary>
+        /// The id of this cluster. Returns "I" if no multi-cluster network is present.
```

Comment:
I think DeploymentId has an Azure-specific meaning. So, I don't think they are the same. ClusterId is a logical name for the cluster (e.g., "uswest", "europe", "front1", "front2"). DeploymentId is something like "a74ff75b9bd6424ebbff3bde9643074b" and changes everytime I redeploy the service.